open-sphere-base/mantle/src/main/java/io/opensphere/mantle/icon/impl/gui/IconManagerFrame.java

```diff
@@ -256,48 +272,47 @@ private void addSelectedToFavorites()
      */
     private void createIconPopupMenuItems(JPopupMenu puMenu)
     {
-        JMenuItem addToFavoritesMI = new JMenuItem("Add Selected To Favorites");
-        addToFavoritesMI.addActionListener(e -> addSelectedToFavorites());
-        puMenu.add(addToFavoritesMI);
-
-        JMenuItem rotateSelectedMI = new JMenuItem("Rotate Selected...");
-        rotateSelectedMI.addActionListener(e -> rotateSelected());
-        puMenu.add(rotateSelectedMI);
+        ArrayList<JMenuItem> menuBarList = getSharedButtons();
+        menuBarList.forEach(button -> puMenu.add(button));
 
-        JMenuItem deleteSelectedMI = new JMenuItem("Delete Selected...");
+        JMenuItem deleteSelectedMI = new JMenuItem("Delete Selected");
         deleteSelectedMI.addActionListener(e -> deleteSelected());
         puMenu.add(deleteSelectedMI);
-
-        JMenuItem deSelectAllMI = new JMenuItem("De-Select All");
-        deSelectAllMI.addActionListener(e -> deselectAll());
-        puMenu.add(deSelectAllMI);
     }
 
     /**
      * Creates the menu bar menu items, adding the newly created items to the
      * {@link #myEditMenu}.
      */
     private void createMenuBarEditMenuItems()
+    {
+        ArrayList<JMenuItem> menuBarList = getSharedButtons();

```

Comment:
You dont need to initialize a list .. just iterate on getShareButtons()

Reply:
fixed